File: samples/c/08objects/triangle.c
#include "triangle.h"

Triangle* createTriangle(float width, float height) {
    Triangle* t = malloc(sizeof(Triangle));
    t->rectangle = createRectangle(width, height);
    t->hypotenuse = sqrt(width * width + height * height);
    return t;
}

float triangleArea(Triangle* t) {
    return rectangleArea(t->rectangle)/2;
}

float trianglePerimeter(Triangle* t) {
    return rectanglePerimeter(t->rectangle)/2 + t->hypotenuse;
}

float deleteTriangle(Triangle* t) {
    deleteRectangle(t->rectangle);
    free(t);
}

File: samples/java/08objects/main/Triangle.java
package main;

import java.lang.Math;
import main.Rectangle;

public class Triangle extends Rectangle {
    private double hypotenuse;

    public Triangle(double width, double height) {
        super(width, height);
        this.hypotenuse = Math.sqrt(width * width + height * height);
    }

    public double perimeter() {
        return super.perimeter() / 2 + this.hypotenuse;
    }

    public double area() {
        return super.area() / 2;
    }

    public double getHypotenuse() {
        return this.hypotenuse;
    }
}
